src/js/structs/ServicePlanPhases.js

```diff
@@ -0,0 +1,38 @@
+import List from './List';
+import ServicePlanPhase from './ServicePlanPhase';
+
+class ServicePlanPhases extends List {
+  constructor() {
+    super(...arguments);
+
+    // Cache lookups
+    this._activePhaseIndex = -1;
+    this._activePhase = null;
+
+    // Find first phase where status is not complete
+    this.getItems().some((phase, phaseIndex) => {
+      if (!(phase instanceof ServicePlanPhase)) {
```

Comment:
we never completely transform the items in `this.list` into `ServicePlanPhase`s anywhere. we just create a phase to check if it's complete